### PR TITLE
Add `zowex uss move` command

### PR DESCRIPTION
**Pull request by zFernand0**
**What It Does**

Introduce the `zowex uss move` command

**How to Test**

`zowex uss move <pathA> <pathB>` 

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

Here are some additional comments and discussion outcomes

**Testing Considerations**:
- Create different types of files: `DIR`, `REG`, `FIFO`, `LNK`, `SOCK`, `CHR`, `BLK`
- Error when moving `DIR` to `REG`: `touch a; mkdir b; mv b a;`
- Avoid moving a `LNK` to its destination: `touch a; ln -s a b; mv b a;`

**Flags**: 
- `--force`: equivalent to z/OSMF's overwrite. Fully implemented! 😋 
- `--follow-links`: Not implemented, but defaults to `true`
  - This could be useful for avoiding some common mistakes with the Unix `mv` command.
  - Right now we follow symbolic links by default, and fail if the source and target are identical.
- `--preserve-attributes`:  idea taken from the copy command/PR but not needed for move

**Notes**:
Here are the result of the various edge cases that came up during a parking lot:
- Should we support USS to/from DS?
  - This is out of scope for now.
  - When we decide to do this, we should test to see if z/OSMF supports it.
- Should we support regular expressions? 
  - This is out of scope for now.
  - When we decide to do this, we should perform a list operation 
- Should we support USS file tagging? 
  - z/OS Unix `mv` options: Binary (`-B`), Text (`-T`), Exec (`-E`).
  - Ignore USS tagging and consider `--preserve` to preserve USS attributes
- Should we follow standard Unix or z/OSMF `mv` behavior?
  - Follow z/OS Unix `mv` behavior
- Should we support interactive `-i` prompts?
  - This didn't come up during discussion, but will treat it as out of scope too

### CODE REVIEW

**Review by jace-roell — COMMENTED** [review]
Spoke offline about hiding the`--force` flag in the help text since it is `true` by default and doesnt take an argument value. Otherwise LGTM 😋

File: native/c/zusf.cpp [hunk 1]
```diff
@@ -924,6 +927,99 @@ int zusf_create_uss_file_or_dir(ZUSF *zusf, const string &file, mode_t mode, boo
...
+  if (target_is_dir)
+  {
+    new_target = zusf_join_path(resolved_target, source.substr(source.find_last_of("/") + 1));
+  }
+
+  if (rename(source.c_str(), new_target.c_str()) == -1)
+  {
+    zusf->diag.e_msg_len = sprintf(zusf->diag.e_msg, "Failed to move file or directory from '%s' to '%s'", source.c_str(), target.c_str());
+    return RTNCD_FAILURE;
+  }
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply native/c/zusf.cpp]: Moved away from rename for now 😢 
I will add the rc of the `zut_run_shell_command` call though 😋

File: native/c/zusf.cpp [hunk 2]
```diff
@@ -924,6 +927,119 @@ int zusf_create_uss_file_or_dir(ZUSF *zusf, const string &file, mode_t mode, boo
...
+  // TODO(zFernand0): Use std::filesystem::absolute instead of realpath when C++17 is available
+  // resolve source path
+  char resolved_source[PATH_MAX];
+  auto absolute_source = realpath(source.c_str(), resolved_source);
+  if (absolute_source == nullptr)
+  {
+    zusf->diag.e_msg_len = sprintf(zusf->diag.e_msg, "Failed to resolve source path '%s'", truncated_source.c_str());
+    return RTNCD_FAILURE;
+  }
+  free(absolute_source);
```
Comment (traeok) [inline native/c/zusf.cpp]: The call to `realpath` returns a pointer to the stack buffer `resolved_source` - it does not `malloc` anything: https://www.ibm.com/docs/en/zos/2.5.0?topic=functions-realpath-resolve-path-name#rpath__title__5

Calling `free` on `absolute_source` is UB because `resolved_source` is stack-allocated - please remove this call. See comment below about second use of `free`

**Review by traeok — DISMISSED** [review]

File: native/c/zusf.cpp [hunk 3]
```diff
@@ -924,6 +927,119 @@ int zusf_create_uss_file_or_dir(ZUSF *zusf, const string &file, mode_t mode, boo
...
+      zusf->diag.e_msg_len = sprintf(zusf->diag.e_msg, "Cannot move pipe '%s'. Target '%s' is not a pipe", truncated_source.c_str(), truncated_target.c_str());
+      return RTNCD_FAILURE;
+    }
+  }
+
+  // handle target being a folder, which means we have to "move" contents around
+  string new_target = target;
+  if (target_is_dir)
+  {
+    new_target = zusf_join_path(resolved_target, source.substr(source.find_last_of("/") + 1));
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply native/c/zusf.cpp]: I forgot to remove it. it was used with the rename approach 😅 

Resolved in 374f2cdd
